### PR TITLE
Loot System Improvements

### PR DESCRIPTION
**Pull request by jk13pclick**
My attempt at #1556. The gist is that more item types can now be used as Loot in the Underground, making Dungeons a more viable strategy for getting rare Items. In general, the goal is to have the intended mechanism (ie: Farm, Underground, etc.) to always be more efficient ways to collect these items, but to allow players other methods to encourage Dungeon grinding for the achievements and for players that really dislike certain mechanisms.

~What is not currently done is adding revised loot for every Dungeon in the game. I thought it prudent to get approval before moving forward + I thought it could be fun to add in some game lore for item locations (ie: Moon Stone in Mt Moon,  Dragon Scale in Mt. Mortar, etc) and could use some input from others + the Discord.~ **This is done now**

As I know far too little about coding and Typescript, this may look super bad and any revisions/improvements would be very welcome. While I worked hard to figure out how to do this, if you don't like it at all then feel free to disregard this PR. I feel that with an aggressive weight curve like the one implemented it won't disrupt the balance of the game too much - hopefully this change could make it into #1571 at the least. Also, if I messed up please let me know, as I mentioned I do not claim to be good at this at all! Let me know what you all think.

### CODE REVIEW

File: src/scripts/dungeons/Dungeon.ts [hunk 1]
```diff
@@ -136,7 +136,7 @@ class Dungeon {
             if (typeof loot === 'string') {
                 return 1;
             } else {
-                return ((<DetailedLoot>loot).weight / (App.game.statistics.dungeonsCleared[GameConstants.getDungeonIndex(DungeonRunner.dungeon.name)]() + 1) + 1) ?? 1;
+                return (10 ^ (<DetailedLoot>loot).weight / (App.game.statistics.dungeonsCleared[GameConstants.getDungeonIndex(DungeonRunner.dungeon.name)]() + 1) + 1) ?? 1;
```
Comment (PixLSteam) [inline src/scripts/dungeons/Dungeon.ts:139]: This might not do what you want it to do. (Hint: ^ is not exponentiation)

Comment (jk13pclick) [reply src/scripts/dungeons/Dungeon.ts:139]: True, I totally forgot about Math.pow!

**Comment by jk13pclick** [comment]
One more tweak to the system, Item Magnet is now nerfed and tied to Dungeon Clears and Weight, would be super broken to have a 50% chance get 2x rare items.

**Comment by jk13pclick** [comment]
One final thought I had while making the loot for all the regions is that this system could be used for some of the currently absent alt forms. Orbs, Drives, etc. could be made into extremely rare loot only obtainable at the end of questlines. Mega Stones could also be obtained using this system (Megas themselves should have their own feature of course).

**Comment by RedSparr0w** [comment]
👍 Nice work, Everything looks good as far as I can tell!
I haven't checked all the items that have been made available, but will trust your judgement on them 😅 

Made a couple of changes:
```
Force all loot to use the new formatting. ({loot: 'name', weight: 4})
Limit minimum weight to original weight specified.
Adjusted rate for Item Magnet based on original weight specified.
Updated description for Item Magnet.
```
Let me know if the changes are all right.

**Comment by RedSparr0w** [comment]
If you want to do them, that's cool, but otherwise I can just merge as is and they can be done in a separate PR?

**Review by RedSparr0w — APPROVED** [review]
Forgot to approve it 👍

**Pull request merged**